### PR TITLE
Custom_Orders - Demo: Strategy setting stoploss/market/take_profit etc on exchnage

### PR DESCRIPTION
**Pull request by creslinux**
(very)Early demonstration of Strategy controlling

- Order Type on the Exchange, Stop / Stop_limit / Market / limit etc
- The stake amount
- The price submitted

Largely the process works by calling money_mgt in strat after buy/sell 
if money_mgt is interested it can populate a list of dicts, where dicts are trades to be made
Money_mgt will then cancel '0' buy/sell =1 and pass the custom_orders list of dicts back through the bot to be processed. 

Demo allows multiple orders to be placed, so 

- laddering buys down to a fib
- Or buying and setting stop-loss in 1 hit
- Or Take profit in stages etc.

Included a strategy demo with stubs for
> limit, market, take_profit and stop_limit orders

Obviously very early code.
Submitted as buys, stops etc are working / showing in GDAX.
We have the order IDs to update trade table.

### CODE REVIEW

**Comment by creslinux** [comment]
If anyone wishes to play with this, please do make use of sandbox api on gdax to not risk real money: 
https://github.com/freqtrade/freqtrade/blob/develop/docs/sandbox-testing.md

Core team please set "do not merge"

**Comment by berlinguyinca** [comment]
love it . Once i figure out the freqcache mac issues here i play with it

**Comment by xmatthias** [comment]
assuming `sell` is triggered and i defined a stoploss-order - wouldn't the bot need to "know" that and cancel that order to sucessfully sell for profit?

otherwise we step over the "i bought 500xxx - and now have open orders for 1000xxx of one asset, 500 in stoploss, and 500 of sell-orders

**Comment by xmatthias** [comment]
I'll close this as i think it was around without activity long enough now.

It'll still be in github's history should we ever need the code again.

**Review by xmatthias — APPROVED** [review]

**Pull request closed without merging**
